src/NServiceBus.AcceptanceTests/Core/OpenTelemetry/When_incoming_message_handled.cs

```diff
@@ -0,0 +1,134 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.OpenTelemetry;
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using AcceptanceTesting;
+using Metrics;
+using NUnit.Framework;
+using Conventions = AcceptanceTesting.Customization.Conventions;
+
+public class When_incoming_message_handled : OpenTelemetryAcceptanceTest
+{
+    static readonly string HandlerTimeMetricName = "nservicebus.messaging.handler_time";
+
+    [Test]
+    public async Task Should_record_success_handling_time()
+    {
+        using TestingMetricListener metricsListener = await WhenMessagesHandled(() => new MyMessage());
+        metricsListener.AssertMetric(HandlerTimeMetricName, 5);
+        AssertMandatoryTags(metricsListener, HandlerTimeMetricName, typeof(MyMessage));
```

Comment:
Is it worth just inlining this? So you don't have to go to the method to see what the mandatory tags are? Again, just making the tags explicit inline, also in the tests

Reply:
If the test fails, it'll tell which are the missing tags. Considering that it's primarily plumbing code, I'd leave it as is.